@j2sol I tried out this patch. The first run was fine, but the second run gives a stack trace because it's looking for update_password in the user dict.

```
- hosts: localhost
  connection: local
  tasks:                                                                                                                                                                                                                                                                            - os_user:
        cloud: local
        state: present
        name: test
        password: secret
        update_password: on_create
        email: thingee@gmail.com
        domain: default
        default_project: demo
```

```
$ ansible-playbook -vvvv test-os_user.yml
Using /home/ubuntu/ansible.cfg as config file
 [WARNING]: Host file not found: hosts

 [WARNING]: provided hosts list is empty, only localhost is available

Loading callback plugin default of type stdout, v2.0 from /home/ubuntu/.virtualenvs/ansible/local/lib/python2.7/site-packages/ansible/plugins/callback/__init__.pyc

PLAYBOOK: test-os_user.yml *****************************************************
1 plays in test-os_user.yml

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
Using module file /home/ubuntu/.virtualenvs/ansible/local/lib/python2.7/site-packages/ansible/modules/core/system/setup.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859 `" && echo ansible-tmp-1478211777.32-196586974776859="` echo $HOME/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmp1TLjm5 TO /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859/setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859/ /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859/setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/ubuntu/.virtualenvs/ansible/bin/python /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859/setup.py; rm -rf "/home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.32-196586974776859/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]

TASK [os_user] *****************************************************************
task path: /home/ubuntu/test-os_user.yml:4
Using module file /home/ubuntu/.virtualenvs/ansible/local/lib/python2.7/site-packages/ansible/modules/core/cloud/openstack/os_user.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715 `" && echo ansible-tmp-1478211777.96-272020615943715="` echo $HOME/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpaEhy_z TO /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715/os_user.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715/ /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715/os_user.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/ubuntu/.virtualenvs/ansible/bin/python /home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715/os_user.py; rm -rf "/home/ubuntu/.ansible/tmp/ansible-tmp-1478211777.96-272020615943715/" > /dev/null 2>&1 && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_hgtdpA/ansible_module_os_user.py", line 273, in <module>
    main()
  File "/tmp/ansible_hgtdpA/ansible_module_os_user.py", line 241, in main
    if _needs_update(params_dict, user):
  File "/tmp/ansible_hgtdpA/ansible_module_os_user.py", line 144, in _needs_update
    if k != 'password' and user[k] != v:
KeyError: 'update_password'

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_name": "os_user"
    },
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_hgtdpA/ansible_module_os_user.py\", line 273, in <module>\n    main()\n  File \"/tmp/ansible_hgtdpA/ansible_module_os_user.py\", line 241, in main\n    if _needs_update(params_dict, user):\n  File \"/tmp/ansible_hgtdpA/ansible_module_os_user.py\", line 144, in _needs_update\n    if k != 'password' and user[k] != v:\nKeyError: 'update_password'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE"
}
        to retry, use: --limit @/home/ubuntu/test-os_user.retry

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1
```

--- a/cloud/openstack/os_user.py
+++ b/cloud/openstack/os_user.py
@@ -43,6 +43,14 @@ options:
         - Password for the user
      required: false
      default: None
+   update_password:
+     required: false
+     default: always
+     choices: ['always', 'on_create']
+     version_added: "2.3"
+     description:
+        - C(always) will attempt to update password.  C(on_create) will only
+          set the password for newly created users.
    email:
      description:
         - Email address for the user
@@ -89,6 +97,17 @@ EXAMPLES = '''
     cloud: mycloud
     state: absent
     name: demouser
+
+# Create a user but don't update password if user exists
+- os_user:
+    cloud: mycloud
+    state: present
+    name: demouser
+    password: secret
+    update_password: on_create
+    email: demo@example.com
+    domain: default
+    default_project: demo
 '''
 
 
@@ -122,12 +141,13 @@ user:
 
 def _needs_update(params_dict, user):
     for k, v in params_dict.items():
-        if k != 'password' and user[k] != v:
+        if k not in ('password', 'update_password') and user[k] != v:
             return True
 
     # We don't get password back in the user object, so assume any supplied
     # password is a change.
-    if params_dict['password'] is not None:
+    if (params_dict['password'] is not None and
+            params_dict['update_password'] == 'always'):
         return True
 
     return False
@@ -164,11 +184,17 @@ def main():
         domain=dict(required=False, default=None),
         enabled=dict(default=True, type='bool'),
         state=dict(default='present', choices=['absent', 'present']),
+        update_password=dict(default='always', choices=['always',
+                                                        'on_create']),
     )
 
     module_kwargs = openstack_module_kwargs()
     module = AnsibleModule(
         argument_spec,
+        required_if=[
+            ('update_password', 'always', ['password']),
+            ('update_password', 'on_create', ['password']),
+        ],
         **module_kwargs)
 
     if not HAS_SHADE:
@@ -181,6 +207,7 @@ def main():
     domain = module.params['domain']
     enabled = module.params['enabled']
     state = module.params['state']
+    update_password = module.params['update_password']
 
     try:
         cloud = shade.openstack_cloud(**module.params)
@@ -203,17 +230,25 @@ def main():
                     enabled=enabled)
                 changed = True
             else:
-                params_dict = {'email': email, 'enabled': enabled, 'password': password}
+                params_dict = {'email': email, 'enabled': enabled,
+                               'password': password,
+                               'update_password': update_password}
                 if domain_id is not None:
                     params_dict['domain_id'] = domain_id
                 if default_project_id is not None:
                     params_dict['default_project_id'] = default_project_id
 
                 if _needs_update(params_dict, user):
-                    user = cloud.update_user(
-                        user['id'], password=password, email=email,
-                        default_project=default_project_id, domain_id=domain_id,
-                        enabled=enabled)
+                    if update_password == 'always':
+                        user = cloud.update_user(
+                            user['id'], password=password, email=email,
+                            default_project=default_project_id,
+                            domain_id=domain_id, enabled=enabled)
+                    else:
+                        user = cloud.update_user(
+                            user['id'], email=email,
+                            default_project=default_project_id,
+                            domain_id=domain_id, enabled=enabled)
                     changed = True
                 else:
                     changed = False

